http/src/test/java/com/redhat/lightblue/client/http/testing/doubles/FakeLightblueRequest.java

```diff
@@ -0,0 +1,31 @@
+package com.redhat.lightblue.client.http.testing.doubles;
+
+import com.redhat.lightblue.client.http.HttpMethod;
+import com.redhat.lightblue.client.request.LightblueRequest;
+
+public class FakeLightblueRequest implements LightblueRequest {
```

Comment:
To the second question, I don't necessarily think it should belong to the lightblue client class. At this point I see the lightblue client class as really just a dumb wrapper around whether or not a request should go to a data node or a metadata node, and turning the String response into a LightblueResponse object. The details of the HTTP request I think should be in the http client (too many "clients"!). I would say since we know Lightblue Rest is configured to compress, then we can allow configuring the clients to accept that encoding as well. The most naive implementation would be a "shouldCompress" flag in the constructor for that `HttpClient` impl, and if this is set, the accept-encoding header is set appropriately, and the http client itself deals with decompressing (it's contract is that it gets to the `LightblueHttpClient` class as a normal uncompressed string). How the http client deals with the encoding would be impl specific.

Do you see issues with that approach?